### PR TITLE
[consensus] fix the leaking quorum cert in consensusdb

### PR DESCRIPTION
**Pull request by zekun000**
## Motivation

This was introduced in https://github.com/libra/libra/pull/2449, we're
not able to prune the last QC it introduced because we don't have
corresponding blocks.

### Have you read the [Contributing Guidelines on pull requests](https://github.com/libra/libra/blob/master/CONTRIBUTING.md#pull-requests)?

Y

## Test Plan

Added asserts.

## Related PRs

(If this PR adds or changes functionality, please take some time to update the docs at https://github.com/libra/website, and link to your PR here.)

### CODE REVIEW

**Comment by zekun000** [comment]
@davidiw the comments are on the function

File: consensus/src/chained_bft/block_storage/sync_manager.rs [hunk 1]
```diff
@@ -187,7 +187,14 @@ impl<T: Payload> BlockStore<T> {
         );
         let mut quorum_certs = vec![];
         quorum_certs.push(highest_commit_cert.clone());
```
Comment (davidiw) [inline consensus/src/chained_bft/block_storage/sync_manager.rs]: what if instead we just took line 190 and deleted 189 and 190-197? and we could add an assert that qc[0] (I think) is the same as highest_commit_cert?

Comment (zekun000) [reply consensus/src/chained_bft/block_storage/sync_manager.rs]: let me assert the id then

Comment (davidiw) [reply consensus/src/chained_bft/block_storage/sync_manager.rs]: wait, I'm confused... your approach match my suggestion either lol.... also you eliminated the assertion that block.len == qc.len
now we're just nitpicking on the code... so I'll accept but I would think... the best solution is....

let quorum_certs = blocks.iter().take(3).map(|b| b.quorum_cert().clone());
assert_eq!(quorum_cert[0], highest_commit_cert);

then you delete a whole lot of code...

Comment (zekun000) [reply consensus/src/chained_bft/block_storage/sync_manager.rs]: your code is not equivalent to the current code
imagine the blocks [C0 <- B1] <- [C1 <- B2] <- [C2 <- B3] <- C3
blocks is [B1, B2, B3],
what we want is [C1, C2, C3], what blocks.iter().take(3).map(|b| b.quorum_cert().clone()) gets is [C0, C1, C2]

**Pull request closed without merging**